src/analysis/seminorm.lean

```diff
@@ -593,6 +618,17 @@ end
 lemma symmetric_ball_zero (r : ℝ) (hx : x ∈ ball p 0 r) : -x ∈ ball p 0 r :=
 balanced_ball_zero p r (-1) (by rw [norm_neg, norm_one]) ⟨x, hx, by rw [neg_smul, one_smul]⟩
 
+@[simp]
+lemma neg_ball (p : seminorm 𝕜 E) (r : ℝ) (x : E) :
+  -ball p x r = ball p (-x) r :=
+by { ext, rw [mem_neg, mem_ball, mem_ball, ←neg_add', sub_neg_eq_add, p.neg], }
+
+@[simp]
+lemma smul_ball_preimage (p : seminorm 𝕜 E) (y : E) (r : ℝ) (a : 𝕜) (ha : a ≠ 0) :
```

Comment:
This should maybe be `smul_preimage_ball`, but you'd have to look for other lemmas about `preimage` to find out whether that's more consistent.

Reply:
I searched mathlib and there is no consistency whatsoever.